### DISCUSSION TITLE
ESP32-Webcam does not compile with additional sensors, or ble

### DISCUSSION THREAD

**Discussion by dilpish** [benzino77/tasmocompiler#288] [General]
Version for ESP32-Webcam does not compile with additional sensors, or ble.

`Compiling .pio/build/tasmota32-webcam/liba60/AT24C256_512/Eeprom24C128_256.cpp.o
/tmp/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_52_3_berry_onewire.ino:26:10: fatal error: OneWire.h: No such file or directory

*****************************************************************`

**Comment by benzino77** [comment]
Hi,

Have a look [here](https://github.com/arendst/Tasmota/blob/e6b4a77abd6621dd49fac12ccfa88cc7f1742983/tasmota/include/tasmota_configurations_ESP32.h#L192)

As you can see Tasmota Dev Team disable I2C, BLE-Bridge or other sensors by default (lines with `#undef` clause) during building generic webcam binary. If you need more information you should ask Tasmota Dev Team on Tasmota discord channel why some features are disabled by default. Those features can be disabled because of compilation problems.

On my side I can tell that I should disable (do not show) feature called _Blootooth_ when esp32-webcam board is choosen - just to not allow user to choose it. To tell the truth probably most sensors groups should be disabled for webcam board because most of them are I2C devices.

**Reply by dilpish** [reply]
I try to complile with some definites: not only i2c. I checked most of the options, and got the following table:

#define USE_DISPLAY - failed
#define USE_WS2812 - failed
#define USE_DS18x20 - failed
#define USE_I2S_AUDIO - failed
#define USE_MI_ESP32 - failed
#define USE_SR04 - failed
#define USE_IR_REMOTE_FULL - failed
#define USE_IR_REMOTE - failed
#define USE_NRF24 - failed
#define USE_RC522 - failed
#define USE_TSL2561 - failed
#define USE_TSL2591 - failed
#define USE_BME68X - failed

But:

#define USE_I2C - SUCCESS 
#define USE_SPI - SUCCESS 
#define USE_HOME_ASSISTANT - SUCCESS   
#define USE_TELEGRAM - SUCCESS   
#define USE_DISCOVERY - SUCCESS   
#define USE_DHT - SUCCESS   
#define USE_LMT01 - SUCCESS   
#define USE_SHT - SUCCESS   
#define USE_BMP - SUCCESS   
#define USE_BH1750 - SUCCESS   
#define USE_SHT3X - SUCCESS   
#define USE_LM75AD - SUCCESS   
#define USE_DHT12 - SUCCESS   
#define USE_AHT1x - SUCCESS   
#define USE_MHZ19 - SUCCESS  
**#define USE_MIBLE - SUCCESS**   
**#define USE_ZIGBEE - SUCCESS** 
#define USE_TCP_BRIDGE - SUCCESS 
**#define USE_ZIGBEE_ZNP **OR** USE_ZIGBEE_EZSP  - SUCCESS** 
**#define USE_IR_RECEIVE - SUCCESS** 
**#define USE_TM1638 - SUCCESS** 
#define USE_HRE - SUCCESS 
**#define USE_DISPLAY_LCD - SUCCESS**
#define USE_DISPLAY_SSD1306 - SUCCESS 
#define USE_DISPLAY_MATRIX - SUCCESS 
#define USE_DISPLAY_SH1106 - SUCCESS 
#define USE_HALLEFFECT - SUCCESS 
**#define USE_IBEACON_ESP32 - SUCCESS** 
#define USE_ETHERNET - SUCCESS

**Reply by benzino77** [reply]
Not sure this is a good idea to do something which is not supported or recommended by TDT. It may or may not work in the future . If you need more information I encourage you to reach TDT on discord channel.

**Reply by roschme** [reply]
Hi,
thanks for this discussion that helped me with the same problem (ESP32 webcam with I2C sensor).

#define USE_I2C
#define USE_BMP

In the custom parameters will do the job.
(unfortunaltly i opened an issue for this, i'm sry for that! understand now that it is a job for TDT)

benzino thank you for your great project!

**Reply by benzino77** [reply]
Happy Tasmotizing ;)

**Reply by rojoricardo** [reply]
Have you found a way to add sensors to esp32-cam or maybe compile a esp32 board with camera and sensors?
I just bought a TTGO T-Cam with mic, and I'd like to compile a tasmota firmware to be able to see the camera, use the display (ssd16... something) use the microphone and the pir sensor..

**Reply by benzino77** [reply]
ESP32-cam board is not intended (from Tasmota point of view) to be used for things other than camera itself. Please search the issues here and on Tasmota GH page to learn more about that.
